### DISCUSSION TITLE
Questions regarding addBase({}) function in config

### DISCUSSION THREAD

**Discussion by BrieucCaillot** [tailwindlabs/tailwindcss#11825] [Help] [answered]
Hi,

I am trying to move out from the @layer base way of doing, for adding style to my headings tags.
Instead I'd like to use the addBase function in the config, or any other way of doing that wouldn't create new styling when using @apply, but instead reusing existing classes.
My questions are : 
- How to use media queries using the addBase method ? 
- How to style as example below my h1 with a class of .heading-title using the addBase method ?

```scss
	h1,
	.h1 {
		@apply text-4xl tracking-tight md:text-5xl leading-12 md:leading-14 lg:text-6xl lg:leading-[4.2rem];
		
		.heading-title &{
			@apply text-3xl md:text-5xl md:leading-14 lg:text-6xl lg:leading-17;
			}
	}
``` 

```typescript
plugins: {
	plugin(function ({ addBase, addComponents, addUtilities, theme }) {
		addBase({
			h1: {
				fontSize: theme('fontSize.4xl'),
				lineHeight: theme('lineHeight.tight'),
				// other properties
				// Would be nice to be able to add classes directly here ?
				class: 'text-4xl tracking-tight md:text-5xl leading-12 md:leading-14 lg:text-6xl lg:leading-[4.2rem]'
			},
		});
	})
}
```

**Comment by wongjn** [comment, marked as answer]
For media queries, have the at rule as the key, and content as an object for that key:

```js
addBase({
  '@media (min-width: 1024px)': {
    // Styles here.
  },
});
```
To use existing classes, you could use `@apply` like:

```js
addBase({
  '@apply text-4xl tracking-tight md:text-5xl': {},
});
```
Though do be aware of Adam Wathan's (creator of Tailwind) recommendations *against* doing this in a Tailwind plugin: https://github.com/tailwindlabs/tailwindcss/discussions/11012#discussioncomment-5653785

**Reply by BrieucCaillot** [reply]
Thanks for you answer, do you know how I could target a heading tag such as h1 with a specific class using addBase ?

**Reply by wongjn** [reply]
Yes, you should be able to use a CSS selector as a string for an object:

```js
addBase({
  'h1.my-class-name': {
    // Styles here.
  },
});
```

**Comment by Foerster64** [comment]
To get started with your first plugin, import Tailwind's plugin function from tailwindcss/plugin . [Valero Credit Card ](https://valerocreditcard.fun/)Then inside your plugins array, call the imported plugin function with an anonymous function as the first argument.
